### PR TITLE
Update API description and Android SDK components for P preview 3.

### PR DESCRIPTION
**Pull request by atsushieno**
The API updates are regeneration from the xamarin-android-docimporter-ng
tool fixes.

### CODE REVIEW

**Comment by atsushieno** [comment]
There is a bunch of removal of `valueOf()` method description, but there is no relevant removal in my tool. My explanation on that is, there used to be embedded valueOf() methods in the stub source code but they have vanished in the latest stubs. They don't affect API anyways.

**Comment by atsushieno** [comment]
Relevant Java.Interop PR: https://github.com/xamarin/java.interop/pull/325

**Comment by jonpryor** [comment]
Are the Java.Interop changes required to merge this PR?

**Comment by jonpryor** [comment]
Why does this PR change e.g. `src/Mono.Android/Profiles/api-24.params.txt`?

I imagine your [earlier comment](https://github.com/xamarin/xamarin-android/pull/1728#issuecomment-391652528) was attempting to address that question, so let me rephrase it (as I'm still confused):

Why is *API-24* (and 25-27) being changed? API-28 would make sense, as API-P is being changed.

Also, why is the subject of this PR "components for P preview 3"? The [Android API-P Overview page](https://developer.android.com/preview/overview) make it look like DP2 is the current version.

**Comment by jonpryor** [comment]
> Why is API-24 (and 25-27) being changed? API-28 would make sense, as API-P is being changed.

The cause for the enumeration `valueOf()` method removals in the `*.params.txt` is because we are [using the Java source stub parser to determine parameter names](https://github.com/xamarin/java.interop/blob/ca27edd23f4dfa2700cf28deaabfe2d74fc72a93/build-tools/xamarin-android-docimporter-ng/generate.sh#L10-L13), and the `valueOf()` method is produced by the Java *compiler*, and is *not* present in Java source code. For example, the `src/android/database/CursorJoiner.java` entry within the `platforms/android-24/android-stubs-src.jar` file contains:

```java
package android.database;
public final class CursorJoiner
  implements java.util.Iterator<android.database.CursorJoiner.Result>, java.lang.Iterable<android.database.CursorJoiner.Result>
{
public static enum Result
{
BOTH(),
LEFT(),
RIGHT();
}
public  CursorJoiner(android.database.Cursor cursorLeft, java.lang.String[] columnNamesLeft, android.database.Cursor cursorRight, java.lang.String[] columnNamesRight) { throw new RuntimeException("Stub!"); }
public  java.util.Iterator<android.database.CursorJoiner.Result> iterator() { throw new RuntimeException("Stub!"); }
public  boolean hasNext() { throw new RuntimeException("Stub!"); }
public  android.database.CursorJoiner.Result next() { throw new RuntimeException("Stub!"); }
public  void remove() { throw new RuntimeException("Stub!"); }
}
```

Note that there is no `valueOf()` method. (Again, *nor should there be*, because the `valueOf()` method is compiler-generated.)

---

What we don't currently know is *why* the `*.params.txt` files *originally* had the `valueOf()` methods on the enum types. Perhaps those were from before we used the Java source parser and were instead parsing documentation?

**Comment by jonpryor** [comment]
It *looks* like the [build error](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/3254/) is ignorable:

```
 Making sure we have http://dl-ssl.google.com/android/repository/build-tools_r24-macosx.zip downloaded and extracted  from it... (TaskId:76)
    Downloading http://dl-ssl.google.com/android/repository/build-tools_r24-macosx.zip into /Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/TestRelease/temp/DesignTimeBuild_False_False_True/Cache/zips (TaskId:76)
/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(439,2): error XA5208: Download failed. Please build again. [/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/TestRelease/temp/DesignTimeBuild_False_False_True/UnnamedProject/UnnamedProject.csproj]
/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/Release/lib/xamarin.android/xbuild/Xamarin/Android/Xamarin.Android.Common.targets(439,2): error XA5208: Reason: The remote server returned an error: (502) Bad Gateway. [/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/TestRelease/temp/DesignTimeBuild_False_False_True/UnnamedProject/UnnamedProject.csproj]
```

This is *probably* ignorable, but just in case, let's try another build...

**Comment by jonpryor** [comment]
build

**Comment by jonpryor** [comment]
As hoped, the [rebuild has no test failures](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/3262/). Instead, it has *different* unrelated (and unreported!) test failures:

```
  mono --debug packages/NUnit.ConsoleRunner.3.7.0/tools/nunit3-console.exe  /Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/external/Java.Interop/build-tools/scripts/../../bin/TestRelease/Java.Interop.Dynamic-Tests.dll  --result="TestResult-Java.Interop.Dynamic-Tests.xml;format=nunit2" --output="bin/TestRelease/TestOutput-Java.Interop.Dynamic-Tests.txt"
...
  Native stacktrace: (TaskId:34)
   (TaskId:34)
        0   mono                                0x000000010d1b12d8 mono_handle_native_crash + 264 (TaskId:34)
        1   mono                                0x000000010d21c896 altstack_handle_and_restore + 70 (TaskId:34)
        2   mono                                0x000000010d27bf09 mono_class_init + 329 (TaskId:34)
        3   mono                                0x000000010d27bf14 mono_class_init + 340 (TaskId:34)
        4   mono                                0x000000010d287dcf mono_class_is_subclass_of + 31 (TaskId:34)
        5   libjava-interop.dylib               0x0000000131ca2968 get_gc_bridge_index + 120 (TaskId:34)
...
  =================================================================
  Got a SIGSEGV while executing native code. This usually indicates
  a fatal error in the mono runtime or one of the native libraries 
  used by your application.
  =================================================================
.../RunNUnitTests.targets(30,5): error MSB3073: The command "mono --debug packages/NUnit.ConsoleRunner.3.7.0/tools/nunit3-console.exe  /Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/external/Java.Interop/build-tools/scripts/../../bin/TestRelease/Java.Interop.Dynamic-Tests.dll  --result="TestResult-Java.Interop.Dynamic-Tests.xml;format=nunit2" --output="bin/TestRelease/TestOutput-Java.Interop.Dynamic-Tests.txt"" exited with code -1.
```
